src/Microsoft.PowerApps.TestEngine/TestInfra/PlaywrightTestInfraFunctions.cs

```diff
@@ -307,21 +307,47 @@ public async Task HandleUserEmailScreen(string selector, string value)
         [ExcludeFromCodeCoverage]
         public async Task HandleUserPasswordScreen(string selector, string value, string desiredUrl)
         {
+            var logger = _singleTestInstanceState.GetLogger();
+
+            // Setting options fot the RunAndWaitForNavigationAsync function
             PageRunAndWaitForNavigationOptions options = new PageRunAndWaitForNavigationOptions();
+
+            // URL that should be redirected to
             options.UrlString = desiredUrl;
-            var logger = _singleTestInstanceState.GetLogger();
+
             ValidatePage();
 
             try
             {
+                // Only continue if redirected to the correct page
                 await Page.RunAndWaitForNavigationAsync(async () =>
                 {
+                    // Find the password box
                     await Page.Locator(selector).WaitForAsync();
+
+                    // Fill in the password
                     await Page.FillAsync(selector, value);
+
+                    // Submit password form
                     await this.ClickAsync("input[type=\"submit\"]");
 
-                    ValidatePage();
-                    await Page.ClickAsync("[id=\"idBtn_Back\"]");
+                    // For instances where there is a 'Stay signed in?' dialogue box.
+                    try
+                    {
+                        // Set options for the WaitForSelectorOptions function
+                        PageWaitForSelectorOptions selectorOptions = new PageWaitForSelectorOptions();
+                        ValidatePage();
+
+                        // Check if we received a 'Stay signed in?' box?
+                        await Page.WaitForSelectorAsync("[id=\"KmsiCheckboxField\"]", selectorOptions);
+
+                        // Click to stay signed in
+                        await Page.ClickAsync("[id=\"idBtn_Back\"]");
+                    }
+                    // If there is no 'Stay signed in?' box, an exception will throw; just catch and continue
+                    catch (Exception)
+                    {
```

Comment:
can you add a debug log for this exception? Just in case there's something else and we need to debug it.

Reply:
The issue is, this will also log if the user put in the wrong password. I had that originally, but since this will also not be present when the user actually does not provide credentials, I removed the logs. If there is an id that's specific to the password page, I can add that check separate and log both.

Reply:
Made changes! :)